mcs/class/corlib/Mono/Runtime.cs

```diff
@@ -79,5 +79,9 @@ public static bool SetGCAllowSynchronousMajor (bool flag)
 			// No longer used
 			return true;
 		}
+
+		[MethodImplAttribute (MethodImplOptions.InternalCall)]
+		static extern void EnableMicrosoftTelemetry (string appBundleID_s, string appSignature_s, string appVersion_s, string merpGUIPath_s);
```

Comment:
Do the `utf16` -> `utf8` conversion here, and pass them to the runtime with `fixed`. You could also check for the Host here to throw a more user-friendly exception rather than asserting in the runtime.

Reply:
Do we have some way of making a utf8 string from `string` other than `Mono.RuntimeStructs.MarshalString` ? That function returns a `Mono.SafeStringMarshal` which assumes that it has ownership of the utf8 string and will free it when ~Dispose~d.  So merp will either need to make a copy, or it will need to be quite explicit about handing off the utf8 string to the runtime.

Reply:
- ifdef it out on MOBILE
- remove `_s` from the names

Reply:
`Encoding.UTF8.GetBytes` is the way to go. You will need to pass the `byte[].Length` to the icall and make sure it reaches `g_strndup` but that should be easy.